python/lsst/meas/extensions/photometryKron/photometryKron.cc

```diff
@@ -0,0 +1,140 @@
+#include "lsst/afw/geom/Point.h"
+#include "lsst/afw/geom/Ellipses.h"
+#include "lsst/afw/geom/AffineTransform.h"
+#include "lsst/afw/image/MaskedImage.h"
+#include "lsst/afw/table/Source.h"
+#include "lsst/meas/base/Algorithm.h"
+#include "lsst/meas/extensions/photometryKron.h"
+
+namespace py = pybind11;
+using namespace pybind11::literals;
+
+namespace lsst {
+namespace meas {
+namespace extensions {
+namespace photometryKron {
+
+namespace {
+
+void declareKronFluxControl(py::module &mod) {
+    py::class_<KronFluxControl> cls(mod, "KronFluxControl");
+
+    cls.def(py::init<>());
+
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, fixed);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, nSigmaForRadius);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, nIterForRadius);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, nRadiusForFlux);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, maxSincRadius);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, minimumRadius);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, enforceMinimumRadius);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, useFootprintRadius);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, smoothingSigma);
+    LSST_DECLARE_CONTROL_FIELD(cls, KronFluxControl, refRadiusName);
+}
+
+void declareKronFluxAlgorithm(py::module &mod) {
+    py::class_<KronFluxAlgorithm, base::SimpleAlgorithm> cls(mod, "KronFluxAlgorithm");
+
+    cls.attr("FAILURE") = py::cast(static_cast<int>(KronFluxAlgorithm::FAILURE));
+    cls.attr("EDGE") = py::cast(static_cast<int>(KronFluxAlgorithm::EDGE));
+    cls.attr("BAD_SHAPE_NO_PSF") = py::cast(static_cast<int>(KronFluxAlgorithm::BAD_SHAPE_NO_PSF));
+    cls.attr("NO_MINIMUM_RADIUS") = py::cast(static_cast<int>(KronFluxAlgorithm::NO_MINIMUM_RADIUS));
+    cls.attr("NO_FALLBACK_RADIUS") = py::cast(static_cast<int>(KronFluxAlgorithm::NO_FALLBACK_RADIUS));
+    cls.attr("BAD_RADIUS") = py::cast(static_cast<int>(KronFluxAlgorithm::BAD_RADIUS));
+    cls.attr("USED_MINIMUM_RADIUS") = py::cast(static_cast<int>(KronFluxAlgorithm::USED_MINIMUM_RADIUS));
+    cls.attr("USED_PSF_RADIUS") = py::cast(static_cast<int>(KronFluxAlgorithm::USED_PSF_RADIUS));
+    cls.attr("SMALL_RADIUS") = py::cast(static_cast<int>(KronFluxAlgorithm::SMALL_RADIUS));
+    cls.attr("BAD_SHAPE") = py::cast(static_cast<int>(KronFluxAlgorithm::BAD_SHAPE));
+    cls.attr("N_FLAGS") = py::cast(static_cast<int>(KronFluxAlgorithm::N_FLAGS));
+
+    cls.def(py::init<KronFluxAlgorithm::Control const &, std::string const &, afw::table::Schema &,
+                     daf::base::PropertySet &>(),
+            "ctrl"_a, "name"_a, "schema"_a, "metadata"_a);
+
+    cls.def("measure", &KronFluxAlgorithm::measure, "measRecord"_a, "exposure"_a);
+    cls.def("measureForced", &KronFluxAlgorithm::measureForced, "measRecord"_a, "exposure"_a, "refRecord"_a,
+            "refWcs"_a);
+    cls.def("fail", &KronFluxAlgorithm::fail, "measRecord"_a, "error"_a = NULL);
```

Comment:
Sorry, I was confusing this with some other code I saw recently that had a 4-parameter version and a 3-parameter version that could trivially have been implemented with the former and a default value for the 4th parameter.

I like your suggestion, but I'd suggest modifying `declareAlgorithm` to not add the constructor(s) at all, and leave that to the caller. It would make the function more generically useful and perhaps eliminate the need for a bit of template metaprogramming, at the expense of requiring a bit more boilerplate for users.

I suggest that be a new ticket, but I can consider doing it on this ticket if you are very keen.